oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -68,7 +68,7 @@
 public class OAuth2Credentials extends Credentials {
 
   private static final long serialVersionUID = 4556936364828217687L;
-  static final Duration DEFAULT_EXPIRATION_MARGIN = Duration.ofMinutes(5);
+  static final Duration DEFAULT_EXPIRATION_MARGIN = Duration.ofMinutes(3).plusSeconds(45);
   static final Duration DEFAULT_REFRESH_MARGIN = Duration.ofMinutes(6);
```

Comment:
I think you want to change both? This to 4 mins or slightly below end expiration to below 3min. 

Right now it will start refreshing at 6 mins and get insta cached response for next 2 mins.